### PR TITLE
reworks posture responses

### PR DESCRIPTION
**Pull request by andrewpmartinez**
- send via bulk updates
- only send on change

### CODE REVIEW

File: ziti/edge/api/client.go [hunk 1]
```diff
@@ -167,11 +169,39 @@ func (c *ctrlClient) CreateSession(svcId string, kind edge.SessionType) (*edge.S
 	return decodeSession(resp)
 }
```
Comment (plorenz) [inline ziti/edge/api/client.go:171]: Do we need to fallback to individual submissions if the endpoint isn't supported? Could remember if you get a 404 and call the individual submission api instead.

Comment (andrewpmartinez) [reply ziti/edge/api/client.go:241]: Sure should and now does.

File: ziti/edge/posture/cache.go [hunk 2]
```diff
@@ -18,231 +18,321 @@ package posture
...
+
+	keys := cache.watchedProcesses.Keys()
+	for _, processPath := range keys {
+		cache.currentData.Processes.Set(processPath, Process(processPath))
 	}
+}
 
-	cache.MacAddresses = MacAddresses()
-	cache.Os = Os()
+// requires both slices to be sorted, returns true if slices are exact equal sets
```
Comment (plorenz) [inline ziti/edge/posture/cache.go:245]: Can use stringz.EqualSlices

**Review by plorenz — APPROVED** [review]

File: ziti/edge/posture/cache.go (hunk 2 above)
Comment (andrewpmartinez) [reply ziti/edge/posture/cache.go]: Done and removed unused function.

**Pull request merged**